c++/src/kj/string.h

```diff
@@ -796,6 +800,19 @@ inline constexpr const char* ConstString::cStr() const { return content == nullp
 inline constexpr size_t String::size() const { return content == nullptr ? 0 : content.size() - 1; }
 inline constexpr size_t ConstString::size() const { return content == nullptr ? 0 : content.size() - 1; }
 
+inline constexpr void String::shrink(size_t size) {
+  if (content != nullptr) {
+    KJ_IREQUIRE(content.size() >= size, "New size must be less than or equal to the current size");
+    content = kj::heapArray(content.first(size));
```

Comment:
Reassigning the content in-place is an unusual semantic for this class. I would expect this to return the new string instead. This would also make it more obvious that the implementation re-allocates the string.